File: server/tests/fixtures/alpha_rule_triggers/java/bug_iteration_modification.java
// Should trigger: bug.iteration_modification
import java.util.*;

public class IterationBug {
    public void removeItems(List<String> items) {
        for (String item : items) {
            if (item.isEmpty()) {
                items.remove(item);  // ConcurrentModificationException!
            }
        }
    }
}

File: server/tests/fixtures/alpha_rule_triggers/csharp/bug_iteration_modification.cs
// Should trigger: bug.iteration_modification
using System.Collections.Generic;

public class IterationBug {
    public void RemoveItems(List<string> items) {
        foreach (var item in items) {
            if (string.IsNullOrEmpty(item)) {
                items.Remove(item);  // InvalidOperationException!
            }
        }
    }
}
